spire/templates/shared-app-redis.yml

```diff
@@ -0,0 +1,107 @@
+
+Parameters:
+  EnvironmentType: { Type: String }
+  RegionMode: { Type: String }
+  RootStackName: { Type: String }
+  RootStackId: { Type: String }
+  SourceSecurityGroupId1: { Type: AWS::EC2::SecurityGroup::Id }
+  VpcId: { Type: AWS::EC2::VPC::Id }
+  VpcPrivateSubnet1Id: { Type: AWS::EC2::Subnet::Id }
+  VpcPrivateSubnet2Id: { Type: AWS::EC2::Subnet::Id }
+  VpcPrivateSubnet3Id: { Type: AWS::EC2::Subnet::Id }
+  StagingInstanceType:
+    Type: String
+    Default: cache.t4g.micro
+  ProductionInstanceType:
+    Type: String
+    Default: cache.t4g.small
+
+Conditions:
+  IsPrimaryRegion: !Equals [!Ref RegionMode, Primary]
+  IsProduction: !Equals [!Ref EnvironmentType, Production]
+
+Resources:
+  SharedAppRedisSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Condition: IsPrimaryRegion
+    Properties:
+      GroupDescription: !Sub Shared ${EnvironmentType} App Redis security group
+      SecurityGroupIngress:
+        - FromPort: 6379
+          IpProtocol: tcp
+          SourceSecurityGroupId: !Ref SourceSecurityGroupId1
+          ToPort: 6379
+      Tags:
+        - { Key: Name, Value: !Sub "${RootStackName}_shared_app_redis" }
+        - { Key: prx:meta:tagging-version, Value: "2021-04-07" }
+        - { Key: prx:cloudformation:stack-name, Value: !Ref AWS::StackName }
+        - { Key: prx:cloudformation:stack-id, Value: !Ref AWS::StackId }
+        - { Key: prx:cloudformation:root-stack-name, Value: !Ref RootStackName }
+        - { Key: prx:cloudformation:root-stack-id, Value: !Ref RootStackId }
+        - { Key: prx:ops:environment, Value: !Ref EnvironmentType }
+        - { Key: prx:dev:application, Value: Common }
+      VpcId: !Ref VpcId
+
+  SharedAppRedisSubnetGroup:
+    Type: AWS::ElastiCache::SubnetGroup
+    Condition: IsPrimaryRegion
+    Properties:
+      Description: !Sub Shared ${EnvironmentType} App Redis subnet group
+      SubnetIds:
+        - !Ref VpcPrivateSubnet1Id
+        - !Ref VpcPrivateSubnet2Id
+        - !Ref VpcPrivateSubnet3Id
+      Tags:
+        - { Key: prx:meta:tagging-version, Value: "2021-04-07" }
+        - { Key: prx:cloudformation:stack-name, Value: !Ref AWS::StackName }
+        - { Key: prx:cloudformation:stack-id, Value: !Ref AWS::StackId }
+        - { Key: prx:cloudformation:root-stack-name, Value: !Ref RootStackName }
+        - { Key: prx:cloudformation:root-stack-id, Value: !Ref RootStackId }
+        - { Key: prx:ops:environment, Value: !Ref EnvironmentType }
+        - { Key: prx:dev:application, Value: Common }
+
+  SharedAppRedisCluster:
+    Type: AWS::ElastiCache::CacheCluster
+    DeletionPolicy: Delete
+    UpdateReplacePolicy: Delete
+    Condition: IsPrimaryRegion
+    Properties:
+      AutoMinorVersionUpgrade: true
+      CacheNodeType: !If
+        - IsProduction
+        - !Ref ProductionInstanceType
+        - !Ref StagingInstanceType
+      CacheSubnetGroupName: !Ref SharedAppRedisSubnetGroup
+      Engine: redis
+      EngineVersion: 6.x
+      NumCacheNodes: 1
```

Comment:
I'm not sure we need > 1 nodes for these?  Augury has certainly been running without it.  We also don't need to back this up, or set any alarms (that I could think of).

Reading the [alarm best practices](https://aws.amazon.com/blogs/database/monitoring-best-practices-with-amazon-elasticache-for-redis-using-amazon-cloudwatch/) ... this work load is fine with evictions, so a memory % alarm doesn't make sense.  Additionally, these newer elasticache instances already reserve memory for system tasks (backups that we don't do, etc).  So I'm not sure we get anything monitoring swap usage.